pkg/action/install.go

```diff
@@ -60,44 +59,47 @@ import (
 // since there can be filepath in front of it.
 const notesFileSuffix = "NOTES.txt"
 
-const defaultDirectoryPermission = 0755
-
 // Install performs an installation operation.
 type Install struct {
 	cfg *Configuration
 
 	ChartPathOptions
 
-	ClientOnly               bool
-	CreateNamespace          bool
-	DryRun                   bool
-	DisableHooks             bool
-	Replace                  bool
-	Wait                     bool
-	WaitForJobs              bool
-	Devel                    bool
-	DependencyUpdate         bool
-	Timeout                  time.Duration
-	Namespace                string
-	ReleaseName              string
-	GenerateName             bool
-	NameTemplate             string
-	Description              string
+	ClientOnly       bool
+	CreateNamespace  bool
+	DryRun           bool
+	DisableHooks     bool
+	Replace          bool
+	Wait             bool
+	WaitForJobs      bool
+	Devel            bool
+	DependencyUpdate bool
+	Timeout          time.Duration
+	Namespace        string
+	ReleaseName      string
+	GenerateName     bool
+	NameTemplate     string
+	Description      string
+	// OutputDir is deprecated because of template-related  code is removed from install action.
+	// Deprecated
```

Comment:
> Removing the template related code from the install action changes the behavior of the SDK for those that import and use it. A behavior change like this breaks semantic versioning.
> 
> The functionality as a whole needs to be deprecated but present during the Helm v3 lifecycle and can functionally change for Helm v4. The Helm policy on backwards compatibility and that we use semantic versions for releases blocks us from being able to merge this change as it is.

But `OutputDir` is used for **template** cmd not install/upgrade . I think install-action writing manifest files is not feature,  If someone truly use it by helm sdk, it won't write all manifest , because hooks relevanted code is in `template.go` .

Reply:
And , if we use `OutputDir` & `IncludeCrds` in install action, it will not write `manifest` and `crds` to buffer which will be finally set in `release.Manifest`, instead ,write it to file. the release generated by this behavior is abnormal ，can not work well in subsequent operations (like uninstall ,upgrade etc.) So I think it is a bug,anyone should never set `OutputDir` or similar fileds as the annotation mentioned.
https://github.com/helm/helm/blob/82a2a2e85ccd9f83455b211ac2a7c7c887ffb9bd/pkg/action/action.go#L184-L216
https://github.com/helm/helm/blob/c2b00659b816a0e4fb9c640a43910387bb8cfdbf/pkg/action/install.go#L259-L263


This PR will not change  the normal behavior of the SDK, beacuse  using  `OutputDir` in install is abnormal and will leading many unforeseen problems.

Reply:
Watting for your advice @mattfarina  :)